packages/core/src/install/createPeerDependencyPatcher.ts

```diff
@@ -21,10 +21,23 @@ export default function (
         if (peerDependencyRules.allowedVersions[peerName] === '*') {
           pkg.peerDependencies![peerName] = '*'
         } else {
-          pkg.peerDependencies![peerName] += ` || ${peerDependencyRules.allowedVersions[peerName]}`
+          const allowedVersions = parseVersions(peerDependencyRules.allowedVersions[peerName])
+          const currentVersions = parseVersions(pkg.peerDependencies![peerName])
```

Comment:
make it a set. It will be faster

```suggestion
          const currentVersions = new Set(parseVersions(pkg.peerDependencies![peerName]))
```

Reply:
While this is generally true for larger datasets or huge numbers of `has` calls, it is not for the sort of small set sizes we're talking about in a peer dependencies array. Here's a benchmark with the proposed change, and at least in my browser it's 48% slower to use a set due to the initialization overhead:

https://jsbench.me/95l20lmjgx/1